create_pdfs.py

```diff
@@ -80,14 +81,38 @@ def main():
 
     certificate_data = []
 
+    course_list = []
     if args.course_id:
-        course_list = [args.course_id]
+        course_id = args.course_id
+        long_course = args.long_course
+        if not long_course:
+            # Default course name to course_id when not supplied
+            # This is helpful when batch testing certs when the name is not
+            # readily available
+            long_course = course_id
+        course_list = [(course_id, args.assign_designation, long_course)]
     else:
-        course_list = settings.CERT_DATA.keys()
-
+        # Compile course list based on openedx-certificates/cert-data.yml
+        # gen_cert.py should and will fail if
```

Comment:
@ggg Based on a convo with Guilio, I was told the cert-templates cert-data.yml DEAFULTS were not allowed to be created by design. 

This came about(including the comment) because the code was failing in gen-cert on line 

184: course_key = CourseKey.from_string(course_id)

due to Course key returning None for default templates when running all cert-data in cert-templates.  I thought this was a bug and fixed the above line to allow DEFUALTS creation.  An unrelated change in retrospect but at the time it was failing due to the sub-template defaults.  Later I reverted based on Guilio's feedback; course_list code should only be run with openedx-certs cert-data.yml.  

The comment is helpful because now if someone tries this again and traces the issue back, they will find the crash is expected.  

If this is confusing or incorrect then we should speak in person.  Maybe I am not quoting this correctly.  That was my take on the conversation.